src/libraries/System.Diagnostics.DiagnosticSource/ref/System.Diagnostics.DiagnosticSourceActivity.cs

```diff
@@ -253,7 +253,25 @@ public sealed class ActivityListener : IDisposable
         public System.Diagnostics.SampleActivity<string>? SampleUsingParentId { get { throw null; } set { throw null; } }
         public System.Diagnostics.SampleActivity<ActivityContext>? Sample { get { throw null; } set { throw null; } }
         public void Dispose() { throw null; }
-   }
+    }
+
+    public delegate bool PropagatorGetterCallback(object carrier, string fieldName, out string? value);
+
+    public abstract class TextMapPropagator
+    {
+        public abstract System.Collections.Generic.IEnumerable<string> Fields { get; }
+        public abstract bool Inject(System.Diagnostics.Activity activity, object carrier, Action<object, string, string> setter);
+        public abstract bool Inject(System.Diagnostics.ActivityContext context, object carrier, Action<object, string, string> setter);
+        public abstract bool Inject(System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, string?>> baggage, object carrier, Action<object, string, string> setter);
+        public abstract bool Extract(object carrier, string fieldName, PropagatorGetterCallback getter, out string? value);
```

Comment:
When would this overload be used?

Reply:
This is needed for asp.net when reading hierarchical ids with the header name `Request-Id`. The other Extract overloads will not be able to read such Id.

Reply:
Thanks. I assume it's not needed for the inject since it can be represented by the Activity?

Reply:
That is right. I am trying to discourage the usage of the hierarchical ids so I wouldn't expose Inject overload for that. I expose the overload that takes Activity object only to support the usage in http client so you don't need to carry any logic there and can just use the propagator as a black box.

Reply:
@MihaZupan could you ping me on teams when you have a minute? Thanks.